recipes/rich-click/meta.yaml

```
@@ -0,0 +1,47 @@
+{% set name = "rich-click" %}
+{% set version = "1.1.1" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/ewels/rich-click/archive/v{{ version }}.tar.gz
+  sha256: 6204e3ae465abca7cf630649d9c9b7853d4ec404c85eeab03b8af77db91616f3
+
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  host:
+    - python
+    - pip
+  run:
+    - python
+    - rich >=10
```

Maybe these should say `>=10,<11` and `>=6?

---

Why less than 11?

---

I mean, `>=10,<unreleased version`. Since you don't actually know whether it is going to break with later versions.

---

So, then `<12` actually. (I had not checked what's the current version.)

---

Ah I see - is that common practice? I don't think that I've ever done that with other packages 🤔 I'd be a bit worried that I'd forget and cause issues for people down the line even if there are no compatibility issues..

---

Since 12 is supposedly going to contain some breaking changes, you can't really know whether rich-click 1.1.1 is going to work with rich 12. Without this pin, people might be able to install a broken version of rich-click in the future and then we have to mark these builds as broken or patch the metadata.

---

Fair enough 👍🏻

In this particular case, @willmcgugan pumps out the major semver versions on a pretty regular basis, which is why I'm a bit nervous (it's very likely that several major versions might pass between any releases of rich-click). But I'll try to remember - I guess I can bump the conda-forge build number and change the requirements if that happens..

```suggestion
    - rich >=10,<12
```

---

> I guess I can bump the conda-forge build number and change the requirements if that happens..

Yes, that's the safe approach :)